compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -15446,7 +15480,7 @@ private STNode parseStmtAsMappingBPOrMappingConsStart(STNode openBrace, STNode m
 
                 // else treat as expression.
                 switchContext(ParserRuleContext.EXPRESSION_STATEMENT);
-                expr = getExpression(bpOrConstructor);
+                expr = getMappingConstructureExpr(bpOrConstructor);
```

Comment:
Would it be possible to incorporate the above change to the existing method?
There are other places that uses the same `getExpression` method, and the issue may also be there for those as well.

Reply:
This is very specific case, we can not use the existing method since `getExpression` is used in other places. If we r to change `QUALIFIED_NAME_REFERENCE` case in `getExpression` It will affect the others

Reply:
Yeah thats true. I was thinking, we could change the way we construct the `MAPPING_BP_OR_MAPPING_CONSTRUCTOR` content, inside `getExpression`

Reply:
Changes done